clients/privacy-center/app/server-environment.ts

```diff
@@ -0,0 +1,206 @@
+  if (typeof window !== "undefined") {
+    throw new Error("Unexpected error, cannot load server environment from client code!");
+  }
+  console.log("Load Privacy Center environment for session..."); // TODO: version number
+
+  // Load environment variables
+  const settings: PrivacyCenterSettings = {
+    FIDES_API_URL: process.env.FIDES_PRIVACY_CENTER__FIDES_API_URL || "http://localhost:8080/api/v1",
+    CONFIG_JSON_URL: process.env.FIDES_PRIVACY_CENTER__CONFIG_JSON_URL || "file:///app/config/config.json",
+    CONFIG_CSS_URL: process.env.FIDES_PRIVACY_CENTER__CONFIG_CSS_URL || "file:///app/config/config.css",
+  };
+
+  // Load configuration file (if it exists)
+  const config = await loadConfigFromFile(settings.CONFIG_JSON_URL);
+
+  // Load styling file (if it exists)
+  const styles = await loadStylesFromFile(settings.CONFIG_CSS_URL);
+  
+  // TODO: Initialize the fides.js file, somehow!
+
+  // Initialize the _environment variable, so usePrivacyCenterEnvironment() can be used
+  _environment = {
+    fidesApiUrl: settings.FIDES_API_URL,
+    config: config,
+    styles: styles,
+  }
+
+  // For backwards-compatibility, override FIDES_API_URL with the value from the config file if present
+  // DEFER: remove backwards compatibility (see https://github.com/ethyca/fides/issues/1264)
+  if (config && (config?.server_url_production || config?.server_url_development || (config as any)?.fidesops_host_production || (config as any)?.fidesops_host_development)) {
+    console.warn(
+      "Using deprecated 'server_url_production' or 'server_url_development' config. " +
+      "Please update to using FIDES_PRIVACY_CENTER__FIDES_API_URL environment variable instead."
+    );
+    const legacyApiUrl =
+      process.env.NODE_ENV === "development" || process.env.NODE_ENV === "test"
+        ? config.server_url_development || (config as any).fidesops_host_development
+        : config.server_url_production || (config as any).fidesops_host_production;
+
+    _environment.fidesApiUrl = legacyApiUrl;
+  }
+
+
+  return _environment;
+}
+
+/**
+ * Load the config.json file from the given URL, or fallback to default filesystem paths.
+ * 
+ * Loading precedence is:
+ * 1) Load from `configJsonUrl` argument
+ * 2) Load from file:///app/config/config.json on filesystem (absolute path)
+ * 3) Load from file:./config/config.json on filesystem (relative path)
+ * 
+ * NOTE: The "/app/config" path is for backwards-compatibility with the initial
+ * version of ethyca/fides-privacy-center, which expected to always load the
+ * configuration file from this well-known path.
+ */
+export const loadConfigFromFile: (configJsonUrl?: string) => Promise<Config | undefined> = async (configJsonUrl?: string) => {
+  const urls = [
+    configJsonUrl,
+    "file:///app/config/config.json",
+    "file:./config/config.json",
+  ];
+  const file = await loadConfigFile(urls);
+  if (file) {
+    const config = JSON.parse(file) as Config;
+    // TODO: validate config?
```

Comment:
Probably should add some defensive checks here to avoid mysterious errors on the client when it tries to use the config.